### PR TITLE
Bugfix: update nginx configuration to adapt to upload large files 

### PR DESCRIPTION
**Pull request by zhou9584**
## Description
image

- [Nginx auto work process configuration](https://ieevee.com/tech/2020/05/11/nginx-process-auto.html)
- disable the Nginx [proxy_buffering](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_buffering)

## Pull Request Checklist

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Code compiles correctly with all tests are passed.
- [x] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [ ] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [ ] Yes
- [ ] No

## How you tested it
*Please make sure the change is tested, you can test it by adding UTs, do local test and share the screenshots, etc.*
upload files with 4 threads and file size > 400MB in each thread
image

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Technical design
- [ ] Build related changes
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe): 

### Feature UI screenshots or Technical design diagrams
*If this is a relatively large or complex change, kick it off by drawing the tech design with PlantUML and explaining why you chose the solution you did and what alternatives you considered, etc...*

### CODE REVIEW

**Review by hydraxman — APPROVED** [review]

**Review by AC-Phoenix — APPROVED** [review]

File: center/deploy_startup/nginx.conf [hunk 1]
```diff
@@ -58,8 +58,13 @@ http {
 
         location  /  {
             proxy_pass http://localhost:9886;
+            proxy_set_header Host $http_host;
+
+            send_timeout  300s;
+            proxy_buffering off;
```
Comment (hydraxman) [inline center/deploy_startup/nginx.conf:64]:
```suggestion
            # When buffering is enabled, nginx receives a response from the proxied server as soon as possible, saving it into the buffers set by the proxy_buffer_size and proxy_buffers directives. More details in https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_buffering
            proxy_buffering off;
```

Comment (hydraxman) [reply center/deploy_startup/nginx.conf:64]: Add some code comments to help understanding.

File: center/deploy_startup/nginx.conf [hunk 2]
```diff
@@ -1,6 +1,6 @@
 
 #user  nobody;
-worker_processes  4;
+worker_processes  auto;
```
Comment (hydraxman) [inline center/deploy_startup/nginx.conf:3]: 问一个问题，如果这里配置成 `auto` ，我们有办法通过日志或者相关输出知道实际的进程数量吗？

我们需要有能力知道，才能知道实际情况是否符合预期。 @zhou9584

Comment (hydraxman) [reply center/deploy_startup/nginx.conf:3]: 因为是在docker环境下，这里会不会有虚拟的CPU数量的影响？

Comment (zhou9584) [reply center/deploy_startup/nginx.conf:3]: we can know this by this command ```ps -ef|grep nginx```
image

Comment (hydraxman) [reply center/deploy_startup/nginx.conf:3]: 把这个命令加到docker的部署脚本里？

**Review by olivershen-wow — APPROVED** [review]

**Pull request merged**